### PR TITLE
feat: create notebook from session replay

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We would like people to be able to create a notebook from a session replay so that they can add a transcription

## Changes

Jul-20-2023 16-45-52

- Show the comment button on a session recording outside of a notebook
- Create a new notebook for the session recording 
- Add autofocus to the editor so that you can start typing immediately
- Create a new concept of target when opening a notebook, that can automatically decide how to open a notebook or force it to open in the sidebar 

## How did you test this code?

Manually for now
Ensured all existing creation flows remained unaffected

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx [hunk 1]
```diff
@@ -61,13 +85,12 @@ export function PlayerMetaLinks(): JSX.Element {
...
 
     const mode = logicProps.mode ?? SessionRecordingPlayerMode.Standard
-    const isInNotebook = !!nodeLogic
 
     return (
         <div className="flex flex-row gap-1 items-center justify-end">
             {![SessionRecordingPlayerMode.Notebook, SessionRecordingPlayerMode.Sharing].includes(mode) ? (
                 <>
-                    {isInNotebook && (
+                    {featureFlags[FEATURE_FLAGS.NOTEBOOKS] && (
```
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx:93]: We have a nifty helper for this called `<FlaggedFeature` which I would recommend using

File: frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx [hunk 2]
```diff
@@ -53,6 +59,24 @@ export function PlayerMetaLinks(): JSX.Element {
                 NotebookNodeType.ReplayTimestamp,
                 buildTimestampCommentContent(currentPlayerTime, sessionRecordingId)
             )
+        } else {
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx:62]: should we wrap this in a feature flag?

**Review by pauldambra — COMMENTED** [review]
Screenshot 2023-07-21 at 10 24 54

It shouldn't have the comment button while in a recording?

I'd say let's wrap this in a flag and then iterate on it?

**Review by benjackwhite — APPROVED** [review]
Generally looks great!

Thoughts that don't have to be in this PR but should be follow up:

1. UX wise its not great that we still have the recording playing outside of the Notebook as well as in the Notebook 🤔 Simple addition would be to pause the recording outside of the notebook or **ideally** to only show the recording outside of the Notebook if it isn't already shown (although this will be tricky)
 
2. Immediately the next thing I would want is to be able to go back to the comments made on a recording when I view it later. Something for the future but we probably want to do something like "show related notebooks" so you can jump to previous comments and not just always create a new notebook each time... 

3. At the very least we should make sure that if you have a notebook open with the same recording in it, then clicking the original comment button adds to that _instead_ of creating a new Notebook... (Solving 1. would make this less important though)

4. When the notebook opens with the recording it's annoying that it starts from 00:00... Would be nice to have it start from the place the comment was added. One option would be to pass a callback into the "createNotebook" thing that can be called once it is successful and then find the recording, jump to the position? Bit messy but I feel like its important to have something work nicely here.

File: frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx (hunk 2 above)
Comment (benjackwhite) [reply frontend/src/scenes/session-recordings/player/PlayerMetaLinks.tsx:62]: I think it is? The button itself I mean

**Pull request merged**